homeassistant/components/mqtt/mixins.py

```diff
@@ -744,7 +744,12 @@ async def async_discovery_update(
                 self.log_name,
                 discovery_hash,
             )
-            await self.async_update(discovery_payload)
+            try:
+                await self.async_update(discovery_payload)
+            except vol.MultipleInvalid as schema_exception:
+                send_discovery_done(self.hass, self._discovery_data)
+                raise schema_exception
```

Comment:
If we're not doing anything with the exception, we don't it in a variable either:
```suggestion
                raise
```

Reply:
for

```python
            try:
                await self.async_update(discovery_payload)
            except vol.Error:
                raise
            finally:
                send_discovery_done(self.hass, self._discovery_data)
```

pylint will give:

```log
************* Module homeassistant.components.mqtt.mixins
homeassistant/components/mqtt/mixins.py:749:12: W0706: The except handler raises immediately (try-except-raise)
```

Reply:
you can simplify that to a `try…finally`

Reply:
That would catch all exceptions, not only the schema errors.

Reply:
May be that is cleaner indeed